llvm/utils/TableGen/Common/CodeGenSchedule.cpp

```diff
@@ -1175,7 +1180,7 @@ void CodeGenSchedModels::collectProcItins() {
     // Insert each itinerary data record in the correct position within
     // the processor model's ItinDefList.
     for (Record *ItinData : ItinRecords) {
-      const Record *ItinDef = ItinData->getValueAsDef("TheClass");
+      Record *ItinDef = ItinData->getValueAsDef("TheClass");
```

Comment:
Do we really have to drop `const`?

Reply:
the `addSchedClass` expects a non const and from everywhere it is used inside it again expects a `non const` .
I tried making everything `const` but the use tree keeps on growing where it expects a `non const` version.